src/cpp/session/modules/SessionConnectionsInstaller.R

```diff
@@ -0,0 +1,476 @@
+#
+# SessionConnectionsInstaller.R
+#
+# Copyright (C) 2009-18 by RStudio, Inc.
+#
+# Unless you have received this program directly from RStudio pursuant
+# to the terms of a commercial license agreement with RStudio, then
+# this program is licensed to you under the terms of version 3 of the
+# GNU Affero General Public License. This program is distributed WITHOUT
+# ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+# MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+# AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+#
+#
+
+.rs.addFunction("odbcBundleOsName", function() {
+   osMapping <- list (
+      linux = "linux",
+      windows = "windows",
+      darwin = "osx"
+   )
+   
+   if (!tolower(Sys.info()["sysname"]) %in% names(osMapping))
+      stop("Operating system \"", Sys.info()["sysname"], "\" is unsupported.")
+   
+   osMapping[[tolower(Sys.info()[["sysname"]])]]
+})
+
+.rs.addFunction("odbcBundleName", function(placeholder) {
+   osName <- .rs.odbcBundleOsName()
+   bitness <- if (grepl("64", Sys.info()["machine"])) "64" else "32"
+   
+   bundleName <- gsub("\\(os\\)", osName, placeholder)
+   bundleName <- gsub("\\(bitness\\)", bitness, bundleName)
+   
+   bundleName
+})
+
+.rs.addFunction("odbcBundleDownload", function(url, placeholder, bundleTemp) {
+   bundleName <- .rs.odbcBundleName(placeholder)
+   bundleUrl <- file.path(url, bundleName)
+   
+   if (exists(bundleTemp)) unlink(bundleTemp, recursive = TRUE)
+   dir.create(bundleTemp, recursive = TRUE)
+   
+   bundleFileTemp <- file.path(bundleTemp, bundleName)
+   download.file(bundleUrl, bundleFileTemp)
+   
+   bundleFileTemp
+})
+
+.rs.addFunction("odbcBundleExtract", function(bundleFileTemp, installPath) {
+   untar(bundleFileTemp, exdir = installPath)
+})
+
+.rs.addFunction("odbcBundleCheckPrereqsUnixodbc", function() {
+   identical(system2("odbcinst", stdout = FALSE), 1L)
+})
+
+.rs.addFunction("odbcBundleCheckPrereqsBrew", function() {
+   identical(system2("brew", stdout = FALSE), 1L)
+})
+
+.rs.addFunction("odbcBundleCheckPrereqsOsx", function() {
+   if (!.rs.odbcBundleCheckPrereqsUnixodbc()) {
+      if (!.rs.odbcBundleCheckPrereqsBrew()) {
+         message("Installing Brew...")
+         system2(
+            "/usr/bin/ruby",
+            args = list(
+               "-e",
+               "\"$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)\""
```

Comment:
It seems a little dangerous to pipe content we don't control from Github into the shell; realize that this is how most people install Homebrew in reality but I'd be pretty surprised if a tool started doing it for me. Do you think it'd be better to ask the user to install Homebrew themselves if it's missing?

Reply:
K, removed.